web/xfa_layer_builder.css

```diff
@@ -157,6 +157,10 @@
   max-height: 100%;
 }
 
+.xfaRight > :is(.xfaRadio, .xfaCheckbox) {
```

Comment:
Don't we need a rule for `.xfaLeft` as well? Moreover, in the Bugzilla ticket the margin directions are reversed, so `.xfaRight` sets a `margin-left` but here we set a `margin-right`; is that correct?

Reply:
The styles in the bugzilla ticket target the `.xfaCaption` `.xfaCaptionForCheckButton` classes which leads to the unexpected html element movement we observed last week (See [calixteman's comment](https://github.com/mozilla/pdf.js/pull/17948#issuecomment-2064902504) re: xfa_issue14071.pdf).

I took an alternative approach by targeting the `.xfaRadio` and `.xfaCheckbox` classes instead and as a result had to swap the margin left/right properties.


The `xfaRight` styles below work correctly and are included in the PR:
```css
.xfaRight > :is(.xfaRadio, .xfaCheckbox) {
  margin-right: 4px;
}
```

The `xfaLeft` styles below do not work correctly and were not included in the PR (I apologize, I should have explained my rationale for this). These styles once again lead to unexpected HTML element movement.
```css
.xfaLeft > :is(.xfaRadio, .xfaCheckbox) {
  margin-left: 4px;
}
```

See the attached examples of incorrect element movement. The green background was add by me to highlight the element shift.

![xfa_issue13994](https://github.com/mozilla/pdf.js/assets/1952835/45a06c6e-a618-4478-bb42-23120b86c95a)
![xfa_issue14150](https://github.com/mozilla/pdf.js/assets/1952835/4abd4b03-dc07-4892-be69-f40a3715378a)
![xfa_issue13213](https://github.com/mozilla/pdf.js/assets/1952835/caf9ff46-de5e-4f53-b431-4ca343936699)

Any suggestions on how to proceed?

Reply:
@Joewebsta Can you share a link to the pdf you used ?

Reply:
The images above were taken from the following test pdfs:
- xfa_issue13994
- xfa_issue14150
- xfa_issue13213

Some additional examples are:
- xfa_bug1716816.pdf
- xfa_bug1716980.pdf
- xfa_bug1718735.pdf
- xfa_dhl_shipment.pdf
- xfa_hsbc_closure.pdf
- xfa_issue13597.pdf